### ISSUE TITLE
Unable to use false as a filter value

### ISSUE THREAD

**Issue by goatfryed** [tannerlinsley/react-table#2519]
**Used version:** 7.2.2

I've a boolean column and implemented a indeterminate checkbox that should filter the column, but it seems that passing false to setFilter sets the filterValue to undefined, therefore removing the filter. I want to use the filterValue false and filter for it.

I haven't found something in the documentation about it, although I can understand that others might expect and use this behavior. If it is intended, we should document it.
Also an option for strict filter mode would be nice, since the value is overridden somewhere in setFilter out of our control.

As a workaround, I'm wrapping my filter value and unwrap it in the filter method for now. Maybe it helps someone

```js
export function indeterminateBooleanFilterMethod(rows, id, filterValue) {
    if (!filterValue) return rows;
    const {wrapped} = filterValue;
    return rows.filter( ({values}) => values[id] === wrapped)
}

export function IndeterminateCheckboxFilter({column: {filterValue, setFilter}}) {

    const {wrapped} = filterValue || {};

    const indeterminateSetter = useCallback(
        el => {
            if (!el) return;
            el.indeterminate = filterValue === undefined;
        },
        [filterValue]
    )

    const onClick = useCallback(
        e => {
            e.stopPropagation();
            if (wrapped === undefined) {
                setFilter({wrapped: true});
                return;
            }
            if (wrapped === true) {
                setFilter({wrapped: false});
                return;
            }
            if (wrapped === false) {
                setFilter(undefined);
            }
        },
        [setFilter, wrapped]
    )

    return <input
        type="checkbox"
        checked={wrapped === true}
        ref={indeterminateSetter}
        onChange={onClick}
    />
}
```

**Comment by tannerlinsley** [comment]
Would you like to submit a PR to fix this?

**Comment by goatfryed** [comment]
Yeah, sure I'll work on a PR.
Should we introduce a flag, in case someone relies on this behaviour?

**Comment by tannerlinsley** [comment]
I’ll consider this a bug. Let’s fix it for real.

**Comment by goatfryed** [comment]
Okay, I've looked into this and noticed, that my bug report is invalid.
In my initial setup, I made a mistake regarding the filter, so it defaulted to text filter, which autoremoves on false values and that makes sense. When I wrapped the filter value, I also fixed the configuration without being aware of it, so it behaved as stated in Examples - Filtering, line 333f to reset on undefined. It wasn't the wrapping that fixed the behavior.

So it's all as intended, if we just use
`
{
  // ...
  filter: 'equals',
}
`
properly as documented or a fitting custom filter. Sorry for wasting your time...

Since I've written a test for custom filters now anyway, I'll add it as a PR. Guess, more tests don't, but I'd understand, if it isn't worth to review it.
